src/com/company/IntListImpl.java

```diff
@@ -0,0 +1,213 @@
+package com.company;
+
+import com.company.exceptions.CheckIndexException;
+import com.company.exceptions.ElementIsNullException;
+import com.company.exceptions.ElementNotFoundException;
+
+public class IntListImpl implements IntList {
+
+    private int[] myIntList = new int[10];
+    private int size = 0;
+
+    @Override
+    public int add(int item) {
+        myIntList[size] = item;
+        size++;
+        grow();
+        return item;
+    }
+
+    @Override
+    public int add(int index, int item) {
+        checkIndex(index);
+        grow();
+        for (int i = size - 1; i >= index; i--) {
+            myIntList[i + 1] = myIntList[i];
+        }
+        myIntList[index] = item;
+        size++;
+        return item;
+    }
+
+    @Override
+    public int set(int index, int item) {
+        checkIndex(index);
+        myIntList[index] = item;
+        return item;
+    }
+
+    @Override
+    public int removeByIndex(int index) {
+        checkIndex(index);
+        int result = myIntList[index];
+        for (int i = index; i < size; i++) {
+            myIntList[i] = myIntList[i + 1];
+        }
+        size--;
```

Comment:
стоит также обнулить последний смещенный элемент, чтобы дублирующее значение не висело в памяти
```suggestion
        myIntList[--size] = null;
```